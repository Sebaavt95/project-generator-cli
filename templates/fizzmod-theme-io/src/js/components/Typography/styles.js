import styled, { css } from 'styled-components';
import { typography } from 'theme';

const setVariant = variant =>
	!variant || !typography[variant] ? typography.body1 : typography[variant];

const Typography = styled.div`
	${({ variant }) => setVariant(variant)};

	${({ capitalize }) =>
		capitalize &&
		css`
			text-transform: lowercase;

			&:first-letter {
				text-transform: capitalize;
			}
		`}
`;

export default { Typography };
