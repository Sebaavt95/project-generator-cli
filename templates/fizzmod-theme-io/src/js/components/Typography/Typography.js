import React from 'react';
import { oneOf, node } from 'prop-types';
import styles from './styles';

const Typography = ({ variant, component, children, ...props }) => {
	const variantMap = {
		h1: 'h1',
		h2: 'h2',
		h3: 'h3',
		h4: 'h4',
		h5: 'h5',
		h6: 'h6',
		subtitle1: 'h6',
		subtitle2: 'h6',
		body1: 'p',
		body2: 'p',
		caption: 'span',
		overline: 'span'
	};

	const componentTag =
		component && variantMap[component] ? variantMap[component] : variantMap[variant];

	return (
		<styles.Typography as={componentTag} {...props} variant={variant}>
			{children}
		</styles.Typography>
	);
};

Typography.propTypes = {
	/** Text */
	children: node.isRequired,
	/** Style from theme.typography */
	variant: oneOf([
		'h1',
		'h2',
		'h3',
		'h4',
		'h5',
		'h6',
		'subtitle1',
		'subtitle2',
		'body1',
		'body2',
		'caption',
		'overline'
	]),
	/** Tag html to render . If component is empty, its render variant prop as tag */
	component: oneOf([
		'h1',
		'h2',
		'h3',
		'h4',
		'h5',
		'h6',
		'subtitle1',
		'subtitle2',
		'body1',
		'body2',
		'caption',
		'overline',
		''
	])
};

Typography.defaultProps = {
	variant: 'body1',
	component: ''
};
export default Typography;
