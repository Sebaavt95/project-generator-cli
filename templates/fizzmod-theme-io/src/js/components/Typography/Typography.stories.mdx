import { Meta, Story, ArgsTable } from '@storybook/addon-docs/blocks';
import Typography from './Typography';

<Meta
	title="Componentes/Typography"
	component={Typography}
	argTypes={{
		component: {
			name: 'Component',
			control: {
				type: 'select',
				options: [
					'h1',
					'h2',
					'h3',
					'h4',
					'h5',
					'h6',
					'subtitle1',
					'subtitle2',
					'body1',
					'body2',
					'caption'
				]
			}
		},
		variant: {
			name: 'Variant',
			control: {
				type: 'select',
				options: [
					'h1',
					'h2',
					'h3',
					'h4',
					'h5',
					'h6',
					'subtitle1',
					'subtitle2',
					'body1',
					'body2',
					'caption'
				]
			}
		}
	}}
/>

# Typography

## Usage

```
import Typography from 'components/Typography';

<Typography>Typography h1</Typography>
<Typography color="primary" variant="h1" component="h2">Typography h1</Typography>
```

## Props

<ArgsTable of={Typography} />

<Story name="Typography h1">
	<Typography variant="h1" component="h1">
		h1. Typograhy
	</Typography>
</Story>

<Story name="Typography h2">
	<Typography variant="h2" component="h2">
		h2. Typograhy
	</Typography>
</Story>

<Story name="Typography h3">
	<Typography variant="h3" component="h3">
		h3. Typograhy
	</Typography>
</Story>

<Story name="Typography h4">
	<Typography variant="h4" component="h4">
		h4. Typograhy
	</Typography>
</Story>

<Story name="Typography h5">
	<Typography variant="h5" component="h5">
		h5. Typograhy
	</Typography>
</Story>

<Story name="Typography h6">
	<Typography variant="h6" component="h6">
		h6. Typograhy
	</Typography>
</Story>

<Story name="Typography subtitle1">
	<Typography variant="subtitle1" component="subtitle1">
		subtitle1 Heading
	</Typography>
</Story>

<Story name="Typography subtitle2">
	<Typography variant="subtitle2" component="subtitle2">
		subtitle2 Heading
	</Typography>
</Story>

<Story name="Typography body1">
	<Typography variant="body1" component="body1">
		body1 Heading
	</Typography>
</Story>

<Story name="Typography body2">
	<Typography variant="body2" component="body2">
		body2 Heading
	</Typography>
</Story>

<Story name="Typography caption">
	<Typography component="caption">caption Typograhy</Typography>
</Story>

## Playground

export const Template = args => <Typography {...args} />;

<Story name="Playground" args={{ children: 'Text', component: 'h1', variant: 'h1' }}>
	{Template.bind({})}
</Story>
