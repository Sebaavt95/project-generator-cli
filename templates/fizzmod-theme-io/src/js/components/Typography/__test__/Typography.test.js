import React from 'react';
import { shallow } from 'enzyme';

import Typography from '../Typography';

describe('Typography component without props', () => {
	it('render', () => {
		const wrapper = shallow(<Typography>typography</Typography>);
		const text = wrapper.find('p').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body1');
		expect(propColor).toEqual('');
	});
});

describe('Typography component with component prop', () => {
	it('render with h1 tag', () => {
		const wrapper = shallow(<Typography component="h1">typography</Typography>);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body1');
		expect(propColor).toEqual('');
	});

	it('render with h1 tag and primary color', () => {
		const wrapper = shallow(
			<Typography color="primary" component="h1">
				typography
			</Typography>
		);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body1');
		expect(propColor).toEqual('primary');
	});

	it('render with h1 tag and secondary color', () => {
		const wrapper = shallow(
			<Typography color="secondary" component="h1">
				typography
			</Typography>
		);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body1');
		expect(propColor).toEqual('secondary');
	});

	it('render with h1 tag and common color red', () => {
		const wrapper = shallow(
			<Typography color="red" component="h1">
				typography
			</Typography>
		);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body1');
		expect(propColor).toEqual('red');
	});
});

describe('Typography component with variant prop', () => {
	it('render with p tag, variant body2', () => {
		const wrapper = shallow(<Typography variant="body2">typography</Typography>);
		const text = wrapper.find('p').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body2');
		expect(propColor).toEqual('');
	});

	it('render with p tag, variant caption', () => {
		const wrapper = shallow(<Typography variant="caption">typography</Typography>);
		const text = wrapper.find('span').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('caption');
		expect(propColor).toEqual('');
	});

	it('render with h6 tag, variant h6', () => {
		const wrapper = shallow(<Typography variant="h6">typography</Typography>);
		const text = wrapper.find('h6').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('h6');
		expect(propColor).toEqual('');
	});
});

describe('Typography component with variant and component prop', () => {
	it('render with h1 tag, variant body2', () => {
		const wrapper = shallow(
			<Typography component="h1" variant="body2">
				typography
			</Typography>
		);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body2');
		expect(propColor).toEqual('');
	});

	it('render with h1 tag, primary color and variant subtitle', () => {
		const wrapper = shallow(
			<Typography color="primary" component="h1" variant="subtitle1">
				typography
			</Typography>
		);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('subtitle1');
		expect(propColor).toEqual('primary');
	});
});

describe('Typography snapshot', () => {
	it('render with h1 tag, variant body2', () => {
		const wrapper = shallow(
			<Typography component="h1" variant="body2" color="secondary">
				typography
			</Typography>
		);
		const text = wrapper.find('h1').text();
		const propVariant = wrapper.props().variant;
		const propColor = wrapper.props().color;
		expect(text).toEqual('typography');
		expect(propVariant).toEqual('body2');
		expect(propColor).toEqual('secondary');
		expect(wrapper).toMatchSnapshot();
	});
});
