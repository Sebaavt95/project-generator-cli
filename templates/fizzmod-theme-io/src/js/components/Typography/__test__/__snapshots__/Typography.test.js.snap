// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography snapshot render with h1 tag, variant body2 1`] = `
<styles__Wrapper
  color="secondary"
  variant="body2"
>
  <h1>
    typography
  </h1>
</styles__Wrapper>
`;
